src/repos/UserRepo.ts

```diff
@@ -75,8 +84,54 @@ const getUsersFollowingPublication = async (pubSlug: string): Promise<User[]> =>
   return users;
 };
 
+/**
+ * Add article to a user's readArticles
+ */
+const appendReadArticle = async (uuid: string, articleID: string): Promise<User> => {
+  const article = await ArticleRepo.getArticleByID(articleID);
+  const user = await UserModel.findOne({ uuid });
+  const checkDuplicates = (prev: boolean, cur: Article) => prev || cur.id === articleID;
+  if (!user) {
+    return user;
+  }
+  if (article) {
```

Comment:
same here, we'd instantiate the article just before this instead of at the beginning of the function where it's just hanging around not doing much till it's used here